app/services/exchange_rates/create_country_currency_history_service.rb

```diff
@@ -0,0 +1,134 @@
+module ExchangeRates
+  class CreateCountryCurrencyHistoryService
+    delegate :client, to: :class
+
+    HOST = 'http://www.hmrc.gov.uk'.freeze
+    YEARS = %w[
+      21
+      22
+      23
+    ].freeze
+    MONTHS = 1..12
+
+    def call
+      ExchangeRateCountryCurrency.truncate
+      ExchangeRateCurrencyRate.by_type('monthly').delete
+
+      YEARS.each do |year|
+        MONTHS.each do |month|
+          month = month.to_s.rjust(2, '0')
+          path = "/softwaredevelopers/rates/exrates-monthly-#{month}#{year}.XML"
+          Rails.logger.info("Getting #{path}")
+          response = client.get(path)
+
+          unless response.status == 200
+            Rails.logger.error("Failed to get #{path}")
+            next
+          end
+
+          xml = Nokogiri::XML(response.body)
+
+          period = xml.xpath('//exchangeRateMonthList').attr('Period').value.split(' to ')
+          period_start_date = Date.parse(period.first)
+          period_end_date = Date.parse(period.last)
+          database_country_currencies = country_currencies_for(period_start_date, period_end_date)
+          xml_country_currencies = xml_country_currencies_for(xml)
+
+          removed, added, changed = enumerate_changes(database_country_currencies, xml_country_currencies)
+          if removed.any?
+            Rails.logger.info("Removing #{removed.length} country currencies #{removed}")
+          end
+          if added.any?
+            Rails.logger.info("Adding #{added.length} country currencies #{added}")
+          end
+          if changed.any?
+            Rails.logger.info("Changing #{changed.length} country currencies #{changed}")
+          end
+
+          File.write("#{year}#{month}.json", JSON.pretty_generate(removed:, added:, changed:))
```

Comment:
Remove me. This is me debugging what has changed month-by-month out of curiosity